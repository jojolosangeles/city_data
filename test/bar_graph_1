set -x
rm actual/bar_graph_1.*
python ../code/main.py scripts/bar_graph_1.txt > actual/bar_graph_1.out
diff expected/bar_graph_1.out actual/bar_graph_1.out
diff expected/bar_graph_1.csv actual
diff -b expected/bar_graph_1.png actual
