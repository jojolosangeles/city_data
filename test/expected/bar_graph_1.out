Test: scripts/bar_graph_1.txt
> # Load collision data into "bar_graph_1" DataFrame
> bar_graph_1 = data/sample_20_Traffic_Collision_Data_from_2010_to_Present.csv
Loaded data/sample_20_Traffic_Collision_Data_from_2010_to_Present.csv into DataFrame bar_graph_1
> 
> # Display the columns loaded (to verify the load was successful)
> bar_graph_1.columns
['DR Number' 'Date Reported' 'Date Occurred' 'Time Occurred' 'Area ID'
 'Area Name' 'Reporting District' 'Crime Code' 'Crime Code Description'
 'MO Codes' 'Victim Age' 'Victim Sex' 'Victim Descent' 'Premise Code'
 'Premise Description' 'Address' 'Cross Street' 'Location']
> 
> # Get rid of the columns we aren't using
> bar_graph_1.drop DR Number, Date Reported, Area ID, Reporting District
Dropped ['DR Number', 'Date Reported', 'Area ID', 'Reporting District'] from DataFrame 'bar_graph_1'
> bar_graph_1.drop Crime Code, Crime Code Description, MO Codes, Premise Code
Dropped ['Crime Code', 'Crime Code Description', 'MO Codes', 'Premise Code'] from DataFrame 'bar_graph_1'
> bar_graph_1.drop Premise Description, Location
Dropped ['Premise Description', 'Location'] from DataFrame 'bar_graph_1'
> 
> # Create some new columns
> bar_graph_1.Date <= [pd.to_datetime(x) for x in dataFrame['Date Occurred']]
> bar_graph_1.Weekday <= [x.weekday() for x in dataFrame['Date']]
> bar_graph_1.Year <= [x.year for x in dataFrame['Date']]
> 
> # Show the current state of the columns (to verify changes)
> bar_graph_1.columns
['Date Occurred' 'Time Occurred' 'Area Name' 'Victim Age' 'Victim Sex'
 'Victim Descent' 'Address' 'Cross Street' 'Date' 'Weekday' 'Year']
> 
> # Save the DataFrame "bar_graph_1" into a CSV file "bar_graph_1.csv"
> bar_graph_1.save
Saved CSV File 'actual/bar_graph_1.csv'
> 
> # Draw a bar graph using "Year" for X axis, and "Number of Accidents" for Y axis
> # Notation for axes is passed to
> bar_graph_1.bar X|Year:O|Year,Y|count()|Number of Accidents
Saved Bar Graph PNG 'actual/bar_graph_1.png'
